submissions/ArthurGorbenko/frogger-game/js/app.js

```diff
@@ -0,0 +1,126 @@
+
+let countForWonRound = 0;
+
+const blockWidth = 100;
+const blockHeight = 90;
+const fieldMaxHeight = 400;
+const fieldMinHeight = -70;
+const fieldMaxWidth = 405;
+const fieldMinWidth = 0;
+const waterLevel = fieldMinHeight + blockHeight;
+const startPositionX = 200;
+const startPositionY = 300;
+
+class Enemy {
+    constructor(x, y) {
+        this.sprite = 'images/enemy-bug.png';
+        this.x = x;
+        this.y = y;
+        this.speed = Math.floor((Math.random() * 200)) + 100;
+    }
+    update(dt) {
+        this.x += this.speed * dt;
+        
+        if (this.x > 600) {
+            this.x = -100;
+        }
+
+        const halfOfSizeImg = 50;
+
+        if (this.x < player.x + halfOfSizeImg &&
+            this.x + halfOfSizeImg > player.x &&
+            this.y < player.y + halfOfSizeImg &&
+            this.y + halfOfSizeImg > player.y) {
+            alert('You are busted!')
+            countForWonRound = 0;
+            player.x = startPositionX;
+            player.y = startPositionY;
+            allEnemies.forEach(el => el.generateSpeed());
+        }
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    generateSpeed(){
+        this.speed = Math.floor((Math.random() * 200)) + 100;
+    }
+}
+
+class Player {
+    constructor(x, y) {
+        this.x = x;
+        this.y = y;
+        this.sprite = "images/char-boy.png";
+
+    }
+    update(keyPressed) {
+        switch (keyPressed) {
+            case "left":
+                if (this.x - blockWidth <= fieldMinWidth) {
+                    return null;
+                } else {
+                    this.x -= blockWidth;
+                    break;
+                }
+            case "up":
+                if (this.y - blockHeight <= fieldMinHeight) {
+                    return null;
+                } else if (this.y - blockHeight < waterLevel) {
+                    countForWonRound++;
+                    if (countForWonRound === 1) {
```

Comment:
in case the player reach the water he starts from column 4, in case player is busted - starts in column 3, 

I believe start should be always the same

https://i.imgur.com/CV9ZEGD.png

Reply:
Done.